src/Core/Configurations/ConfigFileWatcher.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO.Abstractions;
+using Azure.DataApiBuilder.Config;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Path = System.IO.Path;
+
+namespace Azure.DataApiBuilder.Core.Configurations;
+
+public class ConfigFileWatcher
+{
+    private FileSystemWatcher? _fileWatcher;
+    RuntimeConfigProvider? _configProvider;
+
+    public ConfigFileWatcher(RuntimeConfigProvider configProvider)
+    {
+        FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)configProvider.ConfigLoader;
+        string configFileName = loader.ConfigFilePath;
+        IFileSystem fileSystem = (IFileSystem)loader._fileSystem;
+        string? currentDirectoryPath = fileSystem.Directory.GetCurrentDirectory();
+        string configFilePath = Path.Combine(currentDirectoryPath!, configFileName);
+        string path = Path.GetDirectoryName(configFilePath)!;
+        _fileWatcher = new FileSystemWatcher(Path.GetDirectoryName(configFilePath)!)
+        {
+            Filter = Path.GetFileName(configFilePath),
```

Comment:
it can be either, so I will see if we can get a directory from the path provided, in which case we want to use that directory and not do any additional combinations. if no directly is provided than we should get the current directory and combine.